# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsPresContext.cpp
# Commit: 1f5b0b087930
# Full Hash: 1f5b0b0879309d73ed6b0a03bb8872199e2f2162
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 2. Merge FontFaceSetImpl and gfxUserFontSet into a single class. r=emilio
#   
#   This patch merges FontFaceSetImpl and gfxUserFontSet into the same
#   class. This reduces the level of indirection and in theory makes it
#   easier to manage future threading issues.
# ==============================================================================

diff -r aa5527055ad1 -r 1f5b0b087930 layout/base/nsPresContext.cpp
--- a/layout/base/nsPresContext.cpp	Tue Jun 28 15:58:09 2022 +0000
+++ b/layout/base/nsPresContext.cpp	Tue Jun 28 15:58:10 2022 +0000
@@ -207,7 +207,7 @@
   // If there's a user font set, discard any src:local() faces it may have
   // loaded because their font entries may no longer be valid.
   if (auto* fonts = Document()->GetFonts()) {
-    fonts->GetUserFontSet()->ForgetLocalFaces();
+    fonts->GetImpl()->ForgetLocalFaces();
   }
 
   FlushFontCache();